12/umbraco-commerce/reference/storefront-api/README.md

```diff
@@ -0,0 +1,611 @@
+            }
+        }
+    }
+}
+```
+{% endcode %}
+
+## Concepts
+
+Before exploring the API endpoints detailed below, there are a few concepts to keep in mind.
+
+<details>
+
+<summary>Session</summary>
+
+When working with Umbraco Commerce's C# Api, Umbraco Commerce will normally keep track of a series of items for you, such as the current Order ID or the current Language, which it tracks in a cookie. When working in a headless manor however we can no longer rely on cookies for this behaviour and so it becomes the implementors responsibility to remember these items and pass them as Headers to the endpoints that may need to use them for context.
+
+The following is a list of supported headers used for session management:
+
+* `Store` - The ID or Alias of the store the given opperation is being performed against.
+* `Current-Order` - The ID of any current "in-progress" order.
+* `Customer-Reference` - A unique reference for the current customer.
+* `Accept-Language` - The ISO Culture Code of the current front end language.
+* `Currency` - The ID or ISO Code of the current currency. If not supplied, will fallback to the default currency of the default country defined on the store.
+* `Tax-Class` - The ID or Alias of the default tax class. If not supplied, will fallback to the default defined on the store.
+* `Billing-Country` - The ID or ISO Code of the default billing country. If not supplied, falls back to the default country defined on the store.
+* `Billing-Region` - The ID or ISO Code of the default billing region. 
+* `Shipping-Country` - The ID or ISO Code of the default shipping country. If not supplied, falls back to either the supplied billing country or default country defined on the store.
+* `Shipping-Region` - The ID or ISO Code of the default billing region.
+
+</details>
+
+<details>
+
+<summary>Expansion</summary>
+
+**Expansion** allows you to retrive additional data about related entities in the API output for a given resource.
+
+By default, where a resource is connected with another resource, such as an Order holding a reference to it's Currency, the Storefront API will return those connected resources as "Reference" objects which usually only contain the resources ID and Alias/Code. 
+
+```json
+{
+    "cartNumber": "CART-01280-054677-RP4L9",
+    "createDate": "2023-07-03T15:11:17.9220005",
+    "currency": {
+        "$type": "CurrencyRef",
+        "code": "GBP",
+        "id": "30b62176-6a9e-4a51-b2f0-7ce6c80a461a"
+    },
+    "customerInfo": { ... },
+    "discountCodes": [],
+    ...
+}
+```
+
+Should you wish to retrieve the connected resource, you could perform an additional get operation against that resources own endpoint, however this could result in multiple wasteful network requests. To help with this, you can pass an `expand` query parameter with a path to properties you wish to expand, which for those targeted properties will return the full resource object instead of the reference entity.
```

Comment:
```suggestion
Should you wish to retrieve the connected resource, you can perform a `get` operation against that resource's own endpoint. However, this could result in multiple wasteful network requests. Instead, you can pass an `expand` query parameter with a path to properties you wish to expand. For the targeted properties, this query will then return the full resource object instead of the reference entity.
```